spacy/language.py

```diff
@@ -180,7 +176,7 @@ def __init__(
         self.vocab: Vocab = vocab
         if self.lang is None:
             self.lang = self.vocab.lang
-        self._components: List[Tuple[str, "Pipe"]] = []
+        self._components: List[Tuple[str, Callable]] = []
```

Comment:
The `Callable` would have to return a `Doc`, right? Should we type this as `Callable[..., Doc]` then?

Reply:
That's a good point. Then it would be more accurate as: `Callable[[Doc], Doc]`